signing_service/signing_service/signing_service.py

```diff
@@ -105,9 +113,35 @@ def _connect(self, tcp_socket: socket.socket) -> None:
 
     def _handle_connection(self, tcp_socket: socket.socket) -> None:  # pylint: disable=no-self-use
         """ Inner loop that handles data exchange over socket. """
-        while True:
-            data = tcp_socket.recv(1024)
-            tcp_socket.send(data)
+        while not self._was_sigterm_caught():
+            try:
+                deserialized_message = next(
+                    receive_frame(
+                        connection=tcp_socket,
+                        public_key=b'\xf3\x97\x19\xcdX\xda\x86tiP\x1c&\xd39M\x9e\xa4\xddb\x89\xb5,]O\xd5cR\x84\xb85\xed\xc9\xa17e,\xb2s\xeb\n1\xcaN.l\xba\xc3\xb7\xc2\xba\xff\xabN\xde\xb3\x0b\xa6l\xbf6o\x81\xe0;',
+                    )
+                )
+                if isinstance(deserialized_message.payload, Message):
+                    deserialized_message.payload.verify_signature(
+                        public_key=b'\xf3\x97\x19\xcdX\xda\x86tiP\x1c&\xd39M\x9e\xa4\xddb\x89\xb5,]O\xd5cR\x84\xb85\xed\xc9\xa17e,\xb2s\xeb\n1\xcaN.l\xba\xc3\xb7\xc2\xba\xff\xabN\xde\xb3\x0b\xa6l\xbf6o\x81\xe0;',
+                    )
```

Comment:
Doesn't Middleman protocol already check the signature? It looks like something that belongs in there. It knows that it's dealing with golem-messages after all.